pkg/git/git.go

```diff
@@ -250,15 +250,10 @@ func ensureHomeEnvSSHLinkedFromPath(logger *zap.SugaredLogger, homeenv string, h
 	}
 }
 
-func userHasKnownHostsFile(logger *zap.SugaredLogger) (bool, error) {
-	homepath, err := homedir.Dir()
-	if err != nil {
-		logger.Errorf("Unexpected error: getting the user home directory: %v", err)
-		return false, err
-	}
+func userHasKnownHostsFile(homepath string) (bool, error) {
 	f, err := os.Open(filepath.Join(homepath, sshKnownHostsUserPath))
 	if err != nil {
-		if os.IsNotExist(err) {
+		if errors.Is(err, fs.ErrNotExist) {
```

Comment:
I'm not sure we can use `io/fs` because downstream the OpenShift build process is not on a new enough version of `go` yet.

Pinging @vdemeester for confirmation here since it might have changed.

Reply:
Yeah it did not changed 😅 If we merge as-is, we (openshift-pipelines team) would need to caring a patch 😅